src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -581,15 +586,26 @@ private PropertyPatternClauseSyntax ParsePropertyPatternClause()
         private SubpatternSyntax ParseSubpatternElement()
         {
             NameColonSyntax nameColon = null;
-            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken && this.PeekToken(1).Kind == SyntaxKind.ColonToken)
+            ExpressionColonSyntax exprColon = null;
+
+            PatternSyntax pattern = ParsePattern(Precedence.Conditional);
+            // If there is a colon but it's not preceeded by a valid expression, leave it out to parse it as a missing comma, preserving C# 9.0 behavior.
+            if (this.CurrentToken.Kind == SyntaxKind.ColonToken && ConvertPatternToExpressionIfPossible(pattern) is ExpressionSyntax expr)
             {
-                var name = this.ParseIdentifierName();
-                var colon = this.EatToken(SyntaxKind.ColonToken);
-                nameColon = _syntaxFactory.NameColon(name, colon);
+                var colon = EatToken();
+                if (expr is IdentifierNameSyntax identifierName)
+                {
+                    nameColon = _syntaxFactory.NameColon(identifierName, colon);
+                }
+                else
+                {
+                    exprColon = _syntaxFactory.ExpressionColon(expr, colon);
```

Comment:
We should check language version with `IDS_FeatureExtendedPropertyPatterns` (either here or in binding)